src/v2/api/services/ScheduleService.ts

```diff
@@ -94,4 +102,102 @@ export class ScheduleService {
+
+    const difference = Math.ceil((endDate.getTime() - indexEndDate.getTime()) / DAY);
+    const divider = period === Period.EVERY_FORTNIGHT ? 14 : 7;
+    const count = Math.floor(difference / divider);
+
+    const lastEndDate = indexEndDate.setDate(indexEndDate.getDate() + count * divider);
+    return new Date(lastEndDate);
+  }
+
+  private async attachLesson (eventId: string, data: AttachLessonDTO) {
+    let discipline = await this.disciplineRepository.findById(data.disciplineId);
+    if (!discipline.disciplineTypes.some((type) => type.name === data.disciplineType)) {
+      discipline = await this.disciplineRepository.updateById(discipline.id, {
+        disciplineTypes: {
+          create: {
+            name: data.disciplineType,
+          },
+        },
+      });
+    }
+
+    const disciplineType = discipline.disciplineTypes.find(({ name }) => name === data.disciplineType);
+
+    for (const teacherId of data.teachers) {
+      const disciplineTeacher = await this.disciplineTeacherRepository.getOrCreate({ teacherId, disciplineId: discipline.id });
+      if (!disciplineTeacher.roles.some(({ role }) => role === TeacherRoleAdapter[data.disciplineType])) {
+        await this.disciplineTeacherRoleRepository.create({
+          disciplineTeacherId: disciplineTeacher.id,
+          disciplineTypeId: disciplineType.id,
+          role: TeacherRoleAdapter[data.disciplineType],
+        });
+      }
+    }
+
+    return {
+      event: await this.eventRepository.updateById(eventId, {
+        lessons: {
+          create: { disciplineTypeId: disciplineType.id },
+        },
+      }),
+      discipline: await this.disciplineRepository.updateById(data.disciplineId, {
+        description: data.disciplineDescription,
+      }),
+    };
+  }
+
+  async createGroupEvent (groupId: string, body: CreateEventDTO) {
+    await this.checkEventDates(body.startTime, body.endTime);
+    if (body.disciplineId && !body.disciplineType) throw new ObjectIsRequiredException('DisciplineType');
+
+    const endTime = body.period === Period.NO_PERIOD
+      ? body.endTime
+      : await this.getLastEndDate(body.endTime, body.period);
+
+    const endIndex  = await this.getEndIndex(body.startTime, body.endTime, body.period);
+    const eventInfo = [];
+    for (let i = 0; i <= endIndex; i++) {
+      eventInfo.push({ number: i, description: body.eventDescription });
+    }
+
+    const event = await this.eventRepository.create({
+      groupId,
+      name: body.name,
+      period: body.period,
+      startTime: body.startTime,
+      url: body.url,
+      endTime,
+      eventInfo: {
+        createMany: { data: eventInfo },
+      },
+    });
+
+    if (!body.disciplineId) {
+      return {
+        event: { ...event, endTime: body.endTime },
+      };
+    }
+
+    const result = await this.attachLesson(event.id, body as AttachLessonDTO);
+
+    return {
+      event: { ...result.event, endTime: body.endTime },
```

Comment:
why body.endTime and not endTime variable

Reply:
Because that variable contains the last end date, but I need to return only the end of the first pair.